### PR TITLE
Oddities Pack + Hat Fixes

### PR DESCRIPTION
**Pull request by DarkVexon**
Adds the new Oddities pack with some fun Vex cards. Midnight Strike is back!

Also fixes some issues with hat unlocks.
-Fixed hats not unlocking when winning in Act 3 rather than Act 4.
-Fixed mistakenly unlocking all hats rather than not just the hats that were played with.
-Fixed the dropdown not refreshing when hats were unlocked.

### CODE REVIEW

File: src/main/java/thePackmaster/hats/DeathScreenUnlockPatch.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                    SpireAnniversary5Mod.saveUnlockedHats(HatMenu.currentlyUnlockedHats);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+
+                HatMenu.refreshHatDropdown();
+            }
+
+        }
+        return SpireReturn.Continue();
```
Comment (erasels) [inline src/main/java/thePackmaster/hats/DeathScreenUnlockPatch.java:49]: You never use the spire return

File: src/main/java/thePackmaster/hats/DeathScreenUnlockPatch.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+
+    private static class Locator extends SpireInsertLocator {
+        @Override
+        public int[] Locate(CtBehavior ctMethodToPatch) throws Exception {
+            Matcher.MethodCallMatcher methodCallMatcher = new Matcher.MethodCallMatcher(ReturnToMenuButton.class, "hide");
+            int[] lines = LineFinder.findAllInOrder(ctMethodToPatch, methodCallMatcher);
```
Comment (erasels) [inline src/main/java/thePackmaster/hats/DeathScreenUnlockPatch.java:57]: This patches every line that matches, wanted?

File: src/main/java/thePackmaster/patches/odditiespack/PackmasterFoilPatches.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+//            }
+//        }
+//
+//        public static void Postfix(AbstractCard __instance, SpriteBatch sb, boolean hovered, boolean selected) {
+//            if (isFoil(__instance)) {
+//                sb.setShader(oldShader);
+//                sb.setColor(oldColor);
+//            }
+//        }
+//    }
```
Comment (erasels) [inline src/main/java/thePackmaster/patches/odditiespack/PackmasterFoilPatches.java:186]: Maybe remove the commented-out code?

**Review by erasels — CHANGES_REQUESTED** [review]

File: src/main/java/thePackmaster/SpireAnniversary5Mod.java [hunk 4]
```diff
@@ -956,5 +957,29 @@ public void onLoad(String s) {
...
+                        if (foilCards.get(i)) {
+                            if (AbstractDungeon.player.masterDeck.size() > i)
+                                PackmasterFoilPatches.makeFoil(AbstractDungeon.player.masterDeck.group.get(i));
+                        }
+                    }
+            }
+        });
     }
+
+    public static float time = 0f;
```
[the remark this answers is not in the document]
Comment (DarkVexon) [reply src/main/java/thePackmaster/SpireAnniversary5Mod.java:984]: It's for the foil shader (the shiny line going across the card). If that's too intrusive, no problem with removing it

File: src/main/java/thePackmaster/cards/odditiespack/EightBall.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    public EightBall() {
+        super(ID, 1, CardType.ATTACK, CardRarity.SPECIAL, CardTarget.ENEMY, CardColor.COLORLESS);
+        baseDamage = 12;
+        baseBlock = 12;
+    }
+
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        blck();
+        dmg(m, AbstractGameAction.AttackEffect.SLASH_HEAVY);
```
[the remark this answers is not in the document]
Comment (DarkVexon) [reply src/main/java/thePackmaster/cards/odditiespack/EightBall.java:24]: Not anymore! Good call

File: src/main/java/thePackmaster/cards/odditiespack/ZaHando.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                    } else {
+                        att(new ExhaustSpecificCardAction(p.hand.group.get(p.hand.group.size() / 2), p.hand));
+                    }
+                }
+            }
+        });
+    }
+
+    @Override
+    public void hover() {
```
[the remark this answers is not in the document]
Comment (DarkVexon) [reply src/main/java/thePackmaster/cards/odditiespack/ZaHando.java:48]: Just tested, seems to work fine at neow and in Compedium, but I'll change the check to Wiz.isInCombat() just to be sure

File: src/main/java/thePackmaster/patches/odditiespack/PackmasterFoilPatches.java (hunk 3 above)
Comment (DarkVexon) [reply src/main/java/thePackmaster/patches/odditiespack/PackmasterFoilPatches.java:186]: might uncomment it later if I have the time to work through it (tinting the card backs of foil cards)

**Review by erasels — APPROVED** [review]

**Pull request merged**